src/__tests__/Button.test.tsx

```diff
@@ -89,4 +89,51 @@ describe('Button', () => {
     const button = container.getByRole('button')
     expect(button).toMatchSnapshot()
   })
+  it('overrides the styles declared with data attributes with sx prop', () => {
+    const container = render(
+      <>
+        <Button size="small" sx={{fontSize: '32px'}}>
+          Small button
+        </Button>
+        <Button size="large" sx={{height: '20px'}}>
+          Large button
+        </Button>
+        <Button sx={{width: '80%'}} block>
+          Block button
+        </Button>
+        <Button variant="invisible" sx={{color: 'firebrick'}}>
+          Invisible button
+        </Button>
+        <Button leadingIcon={IssueClosedIcon} trailingIcon={SearchIcon} sx={{color: 'purple'}}>
+          Button with visuals
+        </Button>
+        <Button trailingAction={TriangleDownIcon} sx={{color: 'purple'}}>
+          Button with action
+        </Button>
+      </>,
+    )
+    const smallBtn = container.getByRole('button', {name: 'Small button'})
+    // [data-size="small"] sets font-size to 12px in the styles, it should be overrideable with sx prop
+    expect(smallBtn).toHaveStyle('font-size: 32px')
+    const largeBtn = container.getByRole('button', {name: 'Large button'})
+    // [data-size="large"] sets height to 40px in the styles, it should be overrideable with sx prop
+    expect(largeBtn).toHaveStyle('height: 20px')
+    const blockBtn = container.getByRole('button', {name: 'Block button'})
+    // [data-block="block"] sets width to 100% in the styles, it should be overrideable with sx prop
+    expect(blockBtn).toHaveStyle('width: 80%')
+    const noVisualsBtn = container.getByRole('button', {name: 'Invisible button'})
+    // [data-no-visuals] sets color to accent.fg for invisible , it should be overrideable with sx prop
+    expect(noVisualsBtn).toHaveStyle('color: firebrick')
```

Comment:
This test passes even when I comment out [custom sx generater](https://github.com/primer/react/blob/main/src/Button/Button.tsx#L15) which is very odd. Because the same button

```jsx
<Button variant="invisible" sx={{color: 'firebrick'}}>
    Invisible button
</Button>
```

fails to update the color to firebrick when  [custom sx generater](https://github.com/primer/react/blob/main/src/Button/Button.tsx#L15) is commented out. I don't know what is really different in test env than storybook but I added the storybook to compansate this awkward test. Let me know if you have any concerns!